src/game/eop.js

```diff
@@ -116,321 +109,324 @@ export const ElevationOfPrivilege = Game({
-          ...G.threat,
-          modal: !G.threat.modal,
-          new: false,
-          id: threat.id,
-          owner: ctx.playerID,
-          title: threat.title,
-          type: threat.type,
-          severity: threat.severity,
-          description: threat.description,
-          mitigation: threat.mitigation,
-        },
-      }
-    },
-    updateThreat: (G, ctx, field, value) => {
-      return {
-        ...G,
-        threat: {
-          ...G.threat,
-          [field]: value,
+  playerView: PlayerView.STRIP_SECRETS,
+  endIf: (G, ctx) => {
+    if (G.round > G.maxRounds) {
+      let scores = [...G.scores];
+      let winner = 0, max = -1;
+      for (let i=0; i<scores.length; i++) {
+        if (scores[i] > max) {
+          winner = i;
+          max = scores[i];
         }
       }
+      return winner;
+    }
+  },
+  turn: {
+    order: {
+      ...TurnOrder.DEFAULT, // Simple clockwise turns
+      first: function first(G, ctx) { // First player, updated
+        return G.lastWinner;          // every phase (start of game)
+      },
     },
-    selectDiagram: (G, ctx, id) => {
-      // if the player has passed, they shouldn't be able to select diagrams
+    endIf: (G, ctx) => {
       let passed = [...G.passed];
-      if (passed.includes(ctx.playerID)) {
-        return INVALID_MOVE;
-      }
-
-      return {
-        ...G,
-        selectedDiagram: id,
-        selectedComponent: "",
-        selectedThreat: "",
-      }
+      if(passed.length >= ctx.numPlayers) {
+        if(G.numCardsPlayed >= ctx.numPlayers) {
+          //end of trick
+          let lastWinner = getWinner(G.suit, G.dealt);
+          return {next: lastWinner};  // choose next player
+        }
+        return true;
+      };
+      return false;
     },
-    selectComponent: (G, ctx, id) => {
-      // if the player has passed, they shouldn't be able to select components
-      let passed = [...G.passed];
-      if (passed.includes(ctx.playerID)) {
-        return INVALID_MOVE;
-      }
-
-      return {
-        ...G,
-        selectedComponent: id,
-        selectedThreat: "",
+    onEnd: onTurnEnd,
+    stages: {
+      threats: {
+        moves: {
+          addOrUpdateThreat,
+          deleteThreat,
+          pass,
+          selectDiagram,
+          selectComponent,
+          selectThreat,
+          toggleModal,
+          toggleModalUpdate,
+          updateThreat
+        },
       }
     },
-    selectThreat: (G, ctx, id) => {
-      // if the player has passed, they shouldn't be able to select threat
-      let passed = [...G.passed];
-      if (passed.includes(ctx.playerID)) {
-        return INVALID_MOVE;
-      }
+  },
+  
+  moves: {
+    draw,
+    selectDiagram,
+    selectComponent,
+    selectThreat,
+  },
+};
 
-      return {
-        ...G,
-        selectedThreat: id,
-      }
+function toggleModal(G, ctx) {
+  // if the player has passed, they shouldn't be able to toggle the modal
+  let passed = [...G.passed];
```

Comment:
Could you define these and the followings as `const`, please?

Reply:
removed `let passed = [...G.passed];` in favour of a function `hasPlayerPassed(G, ctx)`